<a href="https://colab.research.google.com/github/korakot/sunthorn/blob/master/explore.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

[รวมผลงาน](https://github.com/korakot/sunthorn) กลอน 8 ของสุนทรภู่ 

In [ ]:
!pip install kora -q

In [5]:
import pandas as pd
import kora.data_table  # ให้แสดงตารางสวยๆ

In [2]:
!npx degit korakot/sunthorn -f

npx: installed 1 in 1.018s
> destination directory is not empty. Using --force, continuing
> cloned korakot/sunthorn#master


In [6]:
df = pd.read_csv('all_klon.csv')

In [8]:
print(df.shape)
df.head()  # เริ่มจาก นิราศเมืองแกลง

(131864, 5)


work  line  verse  part                         text
0  nka     1      1     2       โอ้สังเวชวาศนานิจาเอ๋ย
1  nka     2      1     3    จะมีคู่มิได้อยู่ประคองเชย
2  nka     3      1     4       ต้องละเลยดวงใจไว้ไกลตา
3  nka     4      2     1  ถึงทุกข์ใครในโลกที่โศกเศร้า
4  nka     5      2     2     ไม่เหมือนเราภุมรินถวิลหา

In [9]:
# นับจำนวน บท, วรรค
df.groupby('work')[['verse','line']].max()

verse   line
work              
abh   24344  97373
kob    1303   5211
lak    2295   9179
nba     231    923
nin     101    403
nka     248    991
npe     209    835
npt     224    895
npu      88    351
nwa     233    931
ova      76    303
rpi     212    847
sin    3370  13479
sws      36    143

ตัวย่อ | ชื่อเต็ม |  จำนวนบท
-----|-------|--------:
abh  | พระอภัยมณี | 24,344
kob  | โคบุตร    | 1,303
lak  | ลักษณวงศ์  | 2,295
nba  | นิราศพระบาท | 231
nin  | นิราศอิเหนา  | 101
nka  | นิราศเมืองแกลง | 248
npe  | นิราศเมืองเพชร | 209
npt  | นิราศพระประธม | 224
npu  | นิราศภูเขาทอง  | 88
nwa  | นิราศวัดเจ้าฟ้า  | 233
ova  | เพลงยาวถวายโอวาท| 76
rpi  | รำพันพิลาป   | 212
sin  | สิงหไกรภพ   | 3,370
sws  | สวัสดิรักษา   | 36
all  | รวมกลอนทั้งหมด | 32,970

In [10]:
# ดูที่ขึ้นต้นด้วย "โอ้" ทั้งหมด  553 วรรค
df[df.text.str.startswith('โอ้')]

work   line  verse  part                           text
0       nka      1      1     2         โอ้สังเวชวาศนานิจาเอ๋ย
7       nka      8      3     1          โอ้จำใจไกลนุชสุดสวาดิ
22      nka     23      6     4       โอ้อุรารุ่มร้อนอ่อนกำลัง
35      nka     36     10     1           โอ้ธานีศรีอยุธยาเอ๋ย
43      nka     44     12     1     โอ้ดูเดือนเหมือนดวงสุดาแม่
...     ...    ...    ...   ...                            ...
113896  abh  79406  19853     1     โอ้น้องแก้วแววเนตรของเชษฐา
120336  abh  85846  21463     1     โอ้ปานนี้แก้วพี่จะครวญคร่ำ
125656  abh  91166  22793     1  โอ้แก้วเนตรเกศพัฒน์ของพี่เอ๋ย
127313  abh  92823  23207     2    โอ้หลานเอ๋ยโรคย่านี้กล้าหาญ
129734  abh  95244  23812     3        โอ้จันทรจรมาอย่าปรารมภ์

[553 rows x 5 columns]

In [11]:
# ดูที่มีคำว่า "สังเวช"
df[df.text.str.contains('สังเวช')]

work   line  verse  part                        text
0       nka      1      1     2      โอ้สังเวชวาศนานิจาเอ๋ย
738     nka    739    185     4  สังเวชนอนมิใคร่หลับระงับลง
1210    nba    220     56     1     สังเวชวัดธารมาที่อาไศรย
2257    kob    344     87     1       แสนสังเวชนางในใจจะขาด
2407    kob    494    124     3   พระทรงฟังสังเวชพระทัยครัน
...     ...    ...    ...   ...                         ...
119308  abh  84818  21206     1       พระโยคีมีจิตคิดสังเวช
119552  abh  85062  21267     1       เทพเจ้ามีจิตคิดสังเวช
120161  abh  85671  21419     2  คิดสังเวชในพระทัยจึงไต่ถาม
124988  abh  90498  22626     1     พวกนายรองพาณิชคิดสังเวช
131192  abh  96702  24177     1       พระโยคีมีจิตคิดสังเวช

[81 rows x 5 columns]